ui/pages/Report/components/Gregor.jsx

```diff
@@ -1,16 +1,53 @@
 import React from 'react'
-import { Button } from 'semantic-ui-react'
+import { Header } from 'semantic-ui-react'
 
-import { InlineHeader } from 'shared/components/StyledComponents'
+import { validators } from 'shared/components/form/FormHelpers'
+import { ButtonRadioGroup } from 'shared/components/form/Inputs'
+import UploadFormPage from 'shared/components/page/UploadFormPage'
 import { CONSENT_CODES } from 'shared/utils/constants'
+import { HttpRequestHelper } from 'shared/utils/httpRequestHelper'
 
-const Gregor = () => (
-  <div>
-    <InlineHeader size="medium" content="Download GREGoR Reports for Consent Code:" />
-    {CONSENT_CODES.map(
-      code => <Button key={code} content={code} secondary icon="download" as="a" href={`/api/report/gregor/${code}`} />,
-    )}
-  </div>
-)
+const FIELDS = [
+
+  {
+    name: 'deliveryPath',
+    label: 'AnVIL Delivery Bucket Path',
+    placeholder: 'gs:// path',
+    validate: validators.required,
+  },
+  {
+    name: 'consentCode',
+    label: 'Consent Code',
+    component: ButtonRadioGroup,
+    options: CONSENT_CODES.map(code => ({ value: code, text: code })),
+    validate: validators.required,
+  },
+]
+
+class Gregor extends React.PureComponent {
+
+  state = { uploadStats: {} }
+
+  onSubmit = values => new HttpRequestHelper('/api/report/gregor',
+    (uploadStats) => {
+      this.setState({ uploadStats })
+    }, (error) => {
+      if (error?.body) {
+        this.setState({ uploadStats: error.body })
+      }
+      return Promise.reject(error)
+    }).post(values)
+
+  render() {
+    const { uploadStats } = this.state
+    return (
+      <div>
+        <Header size="medium" textAlign="center" content="Validate and Upload GREGoR Reports" />
+        <UploadFormPage fields={FIELDS} uploadStats={uploadStats} onSubmit={this.onSubmit} />
```

Comment:
I wonder if it is okay to say `Upload` because the data doesn't leave the cloud.

Reply:
this is the name of an existing react component that is never exposed to the user and renaming it is out of scope

Reply:
How about the `Upload` in the content prop of the `Header` component in line #45?

Reply:
It uploads the reports to the given AnVIL bucket